en/docs/extended_topics/how_to_convert_image_to_ofrecord.md

```diff
@@ -0,0 +1,147 @@
+
+```shell
+$ wget https://oneflow-static.oss-cn-beijing.aliyuncs.com/oneflow-tutorial-attachments/img2ofrecord.zip
+$ unzip img2ofrecord.zip
+```
+
+- Go to the corresponding directory and run OFRecord production script `img2ofrecord.py`
+
+```shell
+$ cd ./img_to_ofrecord
+$ python img2ofrecord.py --part_num=5 --save_dir=./dataset/ --img_format=.png --image_root=./images/train_set/
+```
+
+- The following output will display as the script runs.
+
+```shell
+The image root is:  ./images/train_set/
+The amount of OFRecord data part is:  5
+The directory of Labels is:  ./images/train_label/label.txt
+The image format is:  .png
+The OFRecord save directory is:  ./dataset/
+Start Processing......
+./images/train_set/00000030_3.png feature saved
+./images/train_set/00000034_0.png feature saved
+./images/train_set/00000026_4.png feature saved
+./images/train_set/00000043_9.png feature saved
+......
+Process image successfully !!!
+```
+
+Thus far, we have created the OFRecord file and saved under `./dataset`.
+
+### Code explanation
+
+The structure of entire code directory shows as follow: 
+
+```
+img_to_ofrecord
+├── images
+	├── train_set
+		├── 00000000_5.png
+		├── 00000001_0.png
+		├── 00000002_4.png
+		......
+	├── train_label
+		├── label.txt
+├── img2ofrecord.py
+├── lenet_train.py
+```
+
+- `images` directory holds the original training dataset and label files.
+
+The label file is stored as `json` here in following format：
+
+```shell
+{"00000030_3.png": 3}
+{"00000034_0.png": 0}
+{"00000026_4.png": 4}
+{"00000043_9.png": 9}
+{"00000047_5.png": 5}
+{"00000003_1.png": 1}
+......
+```
+
+- `img2ofrecord.py` is to covert hand written dataset to OFRecord file or script.
+- `lenet_train.py` is to load the OFRecord dataset we just made and use LeNet model for training. 
+
+Scripts that process image files and convert them to OFRecord format are`img2ofrecord.py`. The command options are as follows：
+
+- `image_root` specify the root directory of the image.
+- `part_num` specify the number of OFRecord files to generate. An error is reported if the number is greater than the total number of images.
+- `label_dir` specify the directory of the label.
+- `img_format` specify the format of the image.
+- `save_dir` specify the directory where the OFRecord file will be saved.
+
+## How OFRecord Reader is encoded
+
+The logic associated with the encoding of OFRecord files is in `img2ofrecord.py`. The encoding process is as follows：
+
+1. Encoding the incoming image data.
+
+```python
+def encode_img_file(filename, ext=".jpg"):
+    img = cv2.imread(filename)
+    encoded_data = cv2.imencode(ext, img)[1]
+    return encoded_data.tostring()
+```
+
+The `ext` is the image encoding format. Currently, OneFlow support the same format of encode and decode for image files as OpenCV. Please refer to [cv::ImwriteFlags](https://docs.opencv.org/3.4/d4/da8/group__imgcodecs.html#ga292d81be8d76901bff7988d18d2b42ac). Which include：
```

Comment:
OneFlow support the same format of encode and decode for image files as OpenCV. -> The format supported by ONEFLOW image encoding and decoding is consistent with that of OpenCV which can be refered in  [cv::ImwriteFlags](https://docs.opencv.org/3.4/d4/da8/group__imgcodecs.html#ga292d81be8d76901bff7988d18d2b42ac).